Lib/unittest/case.py

```diff
@@ -1626,3 +1629,47 @@ def shortDescription(self):
 
     def __str__(self):
         return "{} {}".format(self.test_case, self._subDescription())
+
+
+class _SubTestCleanupHelper:
+    """
+    Helper class to manage cleanups and context managers inside subTest blocks,
+    without exposing full TestCase functionality.
+    """
```

Comment:
Nitpick: Perhaps `Context` would be a better name rather than `Helper`:
```suggestion
class _SubTestContext:
    """
    Helper class to manage cleanups and context managers inside subTest blocks,
    without exposing full TestCase functionality.
    """
```